pkg/blockstorage/zone/zone.go

```diff
@@ -192,13 +194,17 @@ func NodeZonesAndRegion(ctx context.Context, cli kubernetes.Interface) (map[stri
 	if cli == nil {
 		return nil, "", errors.New(nodeZonesErr)
 	}
-	ns, err := cli.CoreV1().Nodes().List(metav1.ListOptions{})
+	ns, err := GetReadySchedulableNodes(cli)
 	if err != nil {
 		return nil, "", errors.Wrap(err, nodeZonesErr)
 	}
 	zoneSet := make(map[string]struct{})
 	regionSet := make(map[string]struct{})
 	for _, n := range ns.Items {
+		// For kubernetes 1.17 onwards failureDomain annotations are being deprecated
+		// and will need to use topology.kubernetes.io/zone=us-east-1c and
+		// topology.kubernetes.io/region=us-east-1
+		// https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#topologykubernetesioregion
```

Comment:
@bathina2 - can you follow up with a PR that does uses the new annotations? The code can look at either for now to maintain compatibility with older versions